karate-core/src/main/java/com/intuit/karate/cucumber/CucumberRunner.java

```diff
@@ -64,23 +70,28 @@ public static KarateStats parallel(Class clazz, int threadCount, String reportDi
             for (int i = 0; i < count; i++) {
                 KarateFeature karateFeature = karateFeatures.get(i);
                 int index = i + 1;                
-                CucumberFeature feature = karateFeature.getFeature();                
-                callables.add(() -> {
-                    // we are now within a separate thread. the reporter filters logs by self thread
-                    String threadName = Thread.currentThread().getName();
-                    KarateReporter reporter = karateFeature.getReporter(reportDir);
-                    KarateRuntime runtime = karateFeature.getRuntime(reporter);
-                    try {                        
-                        feature.run(reporter, reporter, runtime);
-                        logger.info("<<<< feature {} of {} on thread {}: {}", index, count, threadName, feature.getPath());
-                    } catch (Exception e) {
-                        logger.error("karate xml/json generation failed for: {}", feature.getPath());
-                        reporter.setFailureReason(e);
-                    } finally { // try our best to close the report file gracefully so that report generation is not broken
-                        reporter.done();
-                    }
-                    return reporter;
-                });
+                CucumberFeature feature = karateFeature.getFeature();
+
+                filterOnTags(feature);
+
+                if(!feature.getFeatureElements().isEmpty()) {
```

Comment:
somewhere above I think, the KarateStats is given a count of all features that were loaded (after cucumber tag filtering). since this is additional filtering, that count needs to be adjusted